### PR TITLE
Fix NavigationUrl unit tests

### PR DESCRIPTION
**Pull request by hishamco**
>Please change the unit tests, a "/" is not required before a "?" unless the path = ""

@sbwalker you are right about adding `/` if the path = "", but it's required too for non empty path, you could try to run the unit tests

### CODE REVIEW

**Comment by sbwalker** [comment]
When I run the framework and I click on the Private page and then click the Edit Mode icon, the resulting NavigateUrl is "/private?edit=1" and the functionality works as expected. You can see that in this case the Url does not contain a "/" directly before the "?" i( e. "/?" ). So I am not sure how these unit tests pass ( as they contain a contain a "/" directly before the "?" ):

        [InlineData("contoso", "", "pageId=4", "/contoso/?pageId=4")]
        [InlineData("contoso", "", "pageId=4&moduleId=10", "/contoso/?pageId=4&moduleId=10")]

In my interactive testing of the framework, these Urls should be:

        [InlineData("contoso", "", "pageId=4", "/contoso?pageId=4")]
        [InlineData("contoso", "", "pageId=4&moduleId=10", "/contoso?pageId=4&moduleId=10")]

I am very confused how you are getting different results than I am.

**Comment by hishamco** [comment]
Let me clean the solution and build it again ...

**Comment by hishamco** [comment]
I rebuild the solution and run it again, all the tests passed

Tests

@chlupac could please try to run the unit tests for the master branch and let us know if there are failing tests? then try my PR if you don't mind

**Comment by sbwalker** [comment]
I ran the tests and I get:

image

**Comment by sbwalker** [comment]
When I debug the tests I see these 4 are failing:

        [InlineData("", "", "pageId=4", "?pageId=4")]
        [InlineData("contoso", "", "pageId=4&moduleId=10", "/contoso?pageId=4&moduleId=10")]
        [InlineData("", "", "pageId=4&moduleId=10", "?pageId=4&moduleId=10")]
        [InlineData("contoso", "", "pageId=4", "/contoso?pageId=4")]

the results should be:

        [InlineData("", "", "pageId=4", "/?pageId=4")]
        [InlineData("contoso", "", "pageId=4&moduleId=10", "/contoso/?pageId=4&moduleId=10")]
        [InlineData("", "", "pageId=4&moduleId=10", "/?pageId=4&moduleId=10")]
        [InlineData("contoso", "", "pageId=4", "/contoso/?pageId=4")]

**Comment by hishamco** [comment]
Lol

That's what I want to clarify to you from the beginning ;) where `/` is required before querystring in all the cases, and that's what I did in this PR

File: Oqtane.Test/Oqtane.Shared.Tests/UtilitiesTests.cs [hunk 1]
```diff
@@ -8,13 +8,13 @@ public class UtilitiesTests
         [Theory]
         [InlineData("contoso", "login", "returnUrl=/admin", "/contoso/login?returnUrl=/admin")]
         [InlineData("contoso", "admin", "", "/contoso/admin")]
-        [InlineData("contoso", "", "pageId=4", "/contoso?pageId=4")]
-        [InlineData("contoso", "", "pageId=4&moduleId=10", "/contoso?pageId=4&moduleId=10")]
+        [InlineData("contoso", "", "pageId=4", "/contoso/?pageId=4")]
```
Comment (hishamco) [inline Oqtane.Test/Oqtane.Shared.Tests/UtilitiesTests.cs:11]: > When I debug the tests I see these 4 are failing:

    [InlineData("", "", "pageId=4", "?pageId=4")]
    [InlineData("contoso", "", "pageId=4&moduleId=10", "/contoso?pageId=4&moduleId=10")]
    [InlineData("", "", "pageId=4&moduleId=10", "?pageId=4&moduleId=10")]
    [InlineData("contoso", "", "pageId=4", "/contoso?pageId=4")]

> the results should be:

    [InlineData("", "", "pageId=4", "/?pageId=4")]
    [InlineData("contoso", "", "pageId=4&moduleId=10", "/contoso/?pageId=4&moduleId=10")]
    [InlineData("", "", "pageId=4&moduleId=10", "/?pageId=4&moduleId=10")]
    [InlineData("contoso", "", "pageId=4", "/contoso/?pageId=4")]

This is exactly what I did!!

**Comment by sbwalker** [comment]
@hishamco please see #354 . I finally figured out why NavigateUrl was generating inconsistent Urls. It was not handling the case where the alias had a value and path = "". In that case it was adding a "/" which is not necessary. Now the NavigateUrl method is producing consistent results for all combinations of aliases and paths. I updated the tests as well.

**Comment by hishamco** [comment]
I told you before if you want me to modify your proposed implementation to let the unit tests pass without `/` unless the path = ""

**Comment by sbwalker** [comment]
My apologies for the obvious lack of communication in this thread. I am not sure how to prevent this from happening in the future, however I do know that too much time and effort was spent on an item which was never really an issue in the first place.

**Comment by hishamco** [comment]
No problem Shaun .. it happen, please merge your PR, I will close this one ..

**Pull request closed without merging**